3b.run_sam/scripts/1.run_sam2_microscopy.py

```diff
@@ -0,0 +1,615 @@
+                    img = img.convert("L")
+                    img.save(jpeg_file)
+            except Exception as e:
+                print(f"Failed to convert {tiff_file}: {e}")
+
+
+# In[ ]:
+
+
+# get list of dirs in the converted to video dir
+converted_dirs = list(converted_to_video_dir.glob("*"))
+converted_dir_names = [dir for dir in converted_dirs]
+for dir in converted_dir_names:
+    dir = sorted(dir.glob("*.jpeg"))
+    for i in enumerate(dir):
+        # rename the files to be in order
+        i[1].rename(f"{dir[0].parent}/{str(i[0] + 1).zfill(3)}.jpeg")
+
+
+# ### Donwsample each frame to fit the images on the GPU - overwrite the copies JPEGs
+
+# In[ ]:
+
+
+# get files in the directory
+converted_dirs_list = list(converted_to_video_dir.rglob("*"))
+converted_dirs_list = [f for f in converted_dirs_list if f.is_file()]
+# posix path to string
+files = [str(f) for f in converted_dirs_list]
+
+
+# In[ ]:
+
+
+downscale_factor = 12
+# sort the files by name
+# downsample the image
+for f in files:
+    img = io.imread(f)
+    # downsample the image
+    downsampled_img = img[::downscale_factor, ::downscale_factor]
+    # save the downsampled image in place of the original image
+    io.imsave(f, downsampled_img)
+
+
+# ## 3. Get initial masks and centers via StarDist
+
+# ### Get the first frame of each video
+# ### Set up a dict that holds the images path, the first frame_mask, and the first frame_centers
+
+# In[ ]:
+
+
+# where one image set here is a single well and fov over all timepoints
+image_set_dict = {
+    "image_set_name": [],
+    "image_set_path": [],
+    "image_set_first_frame": [],
+    "image_x_y_coords": [],
+    "image_labels": [],
+}
+
+# get the list of directories in the ordered tiffs directory
+dirs = list(converted_to_video_dir.glob("*"))
+dirs = [dir for dir in dirs if dir.is_dir()]
+for dir in dirs:
+    # get the files in the directory
+    files = sorted(dir.glob("*.jpeg"))
+    image_set_dict["image_set_name"].append(dir.name)
+    image_set_dict["image_set_path"].append(str(dir))
+    image_set_dict["image_set_first_frame"].append(files[0])
+
+
+# ### Plot the segementation
+# Plot the following:
+# - the original image
+# - the segmentation
+# - the x,y centers of the segmentation
+# - the extracted masks
+
+# In[ ]:
+
+
+model = StarDist2D.from_pretrained("2D_versatile_fluo")
+
+# loop through each image set and predict the instances
+for i in range(len(image_set_dict["image_set_name"])):
+    print(
+        f"{image_set_dict['image_set_name'][i]}: {image_set_dict['image_set_first_frame'][i]}"
+    )
+    img = io.imread(image_set_dict["image_set_first_frame"][i])
+    labels, _ = model.predict_instances(normalize(img))
+    # convert the labels into position coordinates
+    regions = regionprops(label(labels))
+    coords = np.array([r.centroid for r in regions])
+
+    # plot the points and the masks and the image side by side by side
```

Comment:
To save your computer the hassle, do you need to make a plot for every timepoint in the image set? This seems like it would be most useful for testing.

Reply:
added a if statement for visualization